### ISSUE TITLE
Multithreading is not working

### ISSUE THREAD

**Issue by mr-eyes** [sourmash-bio/pyo3_branchwater#56]
In `manysearch` and probably the rest of the commands, multi-threading is not working. Even when I hard-code the number of cores, it only works on 2. I noticed that with `htop`

**Comment by mr-eyes** [comment]
Maybe we can define it globally in each function

```rust
rayon::ThreadPoolBuilder::new().num_threads(32).build_global().unwrap();
```

**Comment by ctb** [comment]
The benchmarks suggest otherwise? Unless it's broken in the last few days, which is certainly possible... but seems unlikely!

* https://github.com/sourmash-bio/pyo3_branchwater/issues/7
* https://github.com/sourmash-bio/pyo3_branchwater/issues/10
* https://github.com/sourmash-bio/pyo3_branchwater/issues/12
* https://github.com/sourmash-bio/pyo3_branchwater/issues/22

e.g. in the most recent (#22) we see:
```
        Percent of CPU this job got: 3189%
```

with 32 threads.

So! mystery! What system are you running on?

**Comment by mr-eyes** [comment]
I am running on farm. How did you set the `32` cores while `num_threads` does not seem to be passed to rust as per #53 ? Am I getting something wrong?

**Comment by ctb** [comment]
can you try this:
```
export RAYON_THREADS=32
```

**Comment by ctb** [comment]
ref https://github.com/sourmash-bio/pyo3_branchwater/issues/31

**Comment by mr-eyes** [comment]
It still does not work.
Maybe it behaves like that because I have a small number of target sigs? I have 1K query vs. 15 targets.

**Comment by ctb** [comment]
yes, the 15 targets is likely to be a thing!

here's something that runs on farm with the manysearch code, and definitely does do multithreading... 

background repo: https://github.com/dib-lab/2022-branchwater-benchmarking
integrating with pyo3_branchwater: https://github.com/dib-lab/2022-branchwater-benchmarking/pull/8

**Comment by ctb** [comment]
(how does #57 fix this issue? curious what you figured out ;)

**Comment by ctb** [comment]
FYI large scale benchmarks confirm that something is fishy with `manysearch` and threading -

https://github.com/dib-lab/2022-branchwater-benchmarking/issues/9

**Comment by mr-eyes** [comment]
> (how does #57 fix this issue? curious what you figured out ;)

Here're some details
- I started an `srun` job with `2 cpus` but `squeue` showed allocation of `34 cpus` due to `mem<->cpus` proportionality.
- I made an initial code modification to set number of threads, but it was after `current_num_threads` so it showed working in parallel while it's really working on only `2 cpus` not `34`. That's why I was confused.
- The PR does not really fix this issue, but makes sure I get the info about the _actual_ number of cores the program is running on. So eventually fixes this.
